src/main/scala/akka/persistence/cassandra/journal/CassandraJournal.scala

```diff
@@ -1,34 +1,32 @@
 package akka.persistence.cassandra.journal
 
-import java.lang.{ Long => JLong }
+import java.lang.{Long => JLong}
 import java.nio.ByteBuffer
 
-import com.datastax.driver.core.policies.{LoggingRetryPolicy, RetryPolicy}
-import com.datastax.driver.core.policies.RetryPolicy.RetryDecision
-
-import scala.concurrent._
-import scala.collection.immutable.Seq
-import scala.collection.JavaConversions._
-import scala.math.min
-import scala.util.{Success, Failure, Try}
-
-import akka.persistence.journal.AsyncWriteJournal
 import akka.persistence._
 import akka.persistence.cassandra._
+import akka.persistence.journal.AsyncWriteJournal
 import akka.serialization.SerializationExtension
-
 import com.datastax.driver.core._
+import com.datastax.driver.core.policies.RetryPolicy.RetryDecision
+import com.datastax.driver.core.policies.{LoggingRetryPolicy, RetryPolicy}
 import com.datastax.driver.core.utils.Bytes
+import com.typesafe.config.Config
+
+import scala.collection.immutable.Seq
+import scala.concurrent._
+import scala.math.min
+import scala.util.{Failure, Success, Try}
 
-class CassandraJournal extends AsyncWriteJournal with CassandraRecovery with CassandraConfigChecker with CassandraStatements {
+class CassandraJournal(cfg: Config) extends AsyncWriteJournal with CassandraRecovery with CassandraConfigChecker with CassandraStatements {
 
-  val config = new CassandraJournalConfig(context.system.settings.config.getConfig("cassandra-journal"))
+  val config = new CassandraJournalConfig(cfg)
   val serialization = SerializationExtension(context.system)
 
   import config._
 
-  val cluster = clusterBuilder.build
-  val session = cluster.connect()
+  lazy val cluster = clusterBuilder.build
+  lazy val session = cluster.connect()
```

Comment:
+1 for a separate PR. Regarding the race, what about creating the keyspaces/schemas explicitly in the test setup, so that the journal actors are ready to just use it?